circle.yml

```diff
@@ -3,29 +3,46 @@
 machine:
   environment:
     DEPLOY_TOOLKIT: $HOME/deploy_toolkit
+    # Locations of log files
     FAILURE_FILE: /tmp/build_failures.txt
+    DEPENDENCY_LOG:  $CIRCLE_ARTIFACTS/dependency.log
+    TESTING_LOG:  $CIRCLE_ARTIFACTS/testing.log
+    LOG_LEVEL: DEBUG
+    # Used ONLY for experimental branches of deploy_toolkit
+    # Do NOT change this ENV VAR manually!!!!
+    DEPLOY_TOOLKIT_BRANCH: <DEPLOY_TOOLKIT_BRANCH>
+
 dependencies:
   pre:
-    # Need deploy_toolkit before anything else
-    - git clone -b master git@github.com:loanlifecycle/deploy_toolkit.git $DEPLOY_TOOLKIT
-    # These are used by build_notifications.py
-    - pip install paho-mqtt mock nose2 cov-core requests pycurl
+    # Clone loanlifecycle/deploy_toolkit AND allow for testing of experimental branches
+    - |
+      if [ "$DEPLOY_TOOLKIT_BRANCH" == "<DEPLOY_TOOLKIT_BRANCH>" ]; then DEPLOY_TOOLKIT_BRANCH="master"; fi
+      echo "Checking out branch: $DEPLOY_TOOLKIT_BRANCH\n"
+      git clone -b $DEPLOY_TOOLKIT_BRANCH git@github.com:loanlifecycle/deploy_toolkit.git $DEPLOY_TOOLKIT
+    # Install gem bundler
+    - gem install bundler --pre
+    # Update npm binary itself (avoids red warning messages)
+    - npm install -g npm
     - sudo apt-get install --assume-yes --force-yes ruby-dev nodejs npm build-essential
```

Comment:
That would make sense these are ubuntu boxes. you need root privileges to run apt-get on system files. I don't know why you don't need sudo on bundler or npm install other than those might be considered "user" binaries, and not "system" binaries? Not sure I haven't done the homework on that one.